expo/expo
# [expo-updates] Fix asset hash validation

# Why

base64url encoding is case sensitive, so it doesn't make sense to lowercase it when verifying asset hashes.

This will need to be cherry-picked into the sdk-45 branch.

# How

On android, this was broken since only one side of the comparison was being lower cased. The fix was to remove all lower-casing.
On iOS, this was broken for two reasons:
- We weren't verifying it in all the places we download assets
- We were lower casing both sides of the comparison, which although works is technically not correct since there would now be two hashes that are equivalent (though impossible to find).

# Test Plan

1. Breakpoint on both iOS and android in the asset downloading code.
2. Load both an experience without and with expected asset hashes (legacy manifest and new manifest respectively)
3. Ensure validation is run and works.
